zproject/prod_settings_template.py

```diff
@@ -394,7 +397,7 @@
         "url": "{}{}".format("https://", EXTERNAL_HOST),
     },
 }
-SOCIAL_AUTH_SAML_ENABLED_IDPS = {
+SOCIAL_AUTH_SAML_ENABLED_IDPS: Dict[str, "SAMLIdPConfigDict"] = {
```

Comment:
For upgrade reasons, we really don't want to add this type here.  
* When upgrading to this commit from older releases, we'll have to worry about the old templated configuration file not being current.  While in this particular case it should work out OK due to the `TYPE_CHECKING` guard, it's not a good pattern.
* If we later want to change/rename the `SAMLIdPConfigDict`, doing so will break the upgrade process.

I wonder if we can do something in `zproject/configured_settings.py` or `computed_settings` to deal with this.  E.g. we import the parameter in `computed_settings.py` with `from confgured_settings import SOCIAL_AUTH_SAML_ENABLED_IDPS as SOCIAL_AUTH_SAML_ENABLED_IDPS_UNTYPED` and then cast it.

Reply:
Hmm, turns out we don't need that casting trick because mypy correctly chooses the type because we import from default_settings first. So I just changed the prod_template declaration to Dict[str, Any].

```
zproject/computed_settings.py:1120: note: Revealed type is 'builtins.dict[builtins.str, TypedDict('zerver.lib.types.SAMLIdPConfigDict',
```